File: 219 Contains Duplicate II/Python/Solution.py
from typing import Dict, List, Self


class Solution:
    def containsNearbyDuplicate(self: Self, nums: List[int], k: int) -> bool:
        numToIdx: Dict[int, int] = {}
        for i in range(len(nums)):
            num: int = nums[i]
            if num in numToIdx and (i - numToIdx[num]) <= k:
                return True
            numToIdx[num] = i
        return False

File: 219 Contains Duplicate II/Java/Solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> numToIdx = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (numToIdx.containsKey(num) && (i - numToIdx.get(num)) <= k) {
                return true;
            }
            numToIdx.put(num, i);
        }

        return false;
    }
}